Bugzilla/Search/Quicksearch.pm

```diff
@@ -477,6 +487,9 @@ sub _handle_field_names {
         if ($translated eq 'bug_status' || $translated eq 'resolution') {
           $bug_status_set = 1;
         }
+        if ($translated =~ /^(?:classification|product|component)$/) {
+          $bug_product_set = 1;
```

Comment:
Shouldn't the hook be called *after* this is set?

Reply:
If I read the code correctly, the hook is called properly with `bug_product_set`. Tested locally and it worked.